assets/extensions/all-extensions.js

```diff
@@ -1,73 +1,292 @@
+		columns: 4,
+		type: 'small-list',
+		title: 'Learner Engagement',
+		items: [
+			{
+				key: 'sensei-certificates',
+				extensionSlug: 'sensei-certificates',
+			},
+			{
+				key: 'sensei-share-your-grade',
+				extensionSlug: 'sensei-share-your-grade',
+			},
+			{
+				key: 'sensei-post-to-course',
+				extensionSlug: 'sensei-post-to-course',
+			},
+		],
+	},
+	{
+		key: 'grid-example',
+		columns: 12,
+		type: 'grid-list',
+		items: [
+			{
+				key: 'sensei-share-your-grade',
+				extensionSlug: 'sensei-share-your-grade',
+			},
+			{
+				key: 'sensei-post-to-course',
+				extensionSlug: 'sensei-post-to-course',
+			},
+			{
+				key: 'sensei-media-attachments',
+				extensionSlug: 'sensei-media-attachments',
+			},
+			{
+				key: 'sensei-course-participants',
+				extensionSlug: 'sensei-course-participants',
+			},
+			{
+				key: 'sensei-course-progress',
+				extensionSlug: 'sensei-course-progress',
+			},
+		],
+	},
+	{
+		key: 'inner-sections-example',
+		columns: 6,
+		title: 'Inner Sections Example',
+		innerSections: [
+			{
+				key: 'sub-section',
+				columns: 12,
+				type: 'small-list',
+				title: 'Sub Section',
+				items: [
+					{
+						key: 'sensei-share-your-grade',
+						extensionSlug: 'sensei-share-your-grade',
+					},
+					{
+						key: 'sensei-post-to-course',
+						extensionSlug: 'sensei-post-to-course',
+					},
+				],
+			},
+			{
+				key: 'sub-section-2',
+				columns: 12,
+				type: 'small-list',
+				title: 'Sub Section 2',
+				items: [
+					{
+						key: 'sensei-post-to-course',
+						extensionSlug: 'sensei-post-to-course',
+					},
+				],
+			},
+		],
+	},
+	{
+		key: 'inner-sections-example2',
+		columns: 6,
+		innerSections: [
+			{
+				key: 'sub-section-3',
+				columns: 12,
+				type: 'small-list',
+				items: [
+					{
+						key: 'sensei-post-to-course',
+						extensionSlug: 'sensei-post-to-course',
+					},
+				],
+			},
+			{
+				key: 'sub-section-4',
+				columns: 12,
+				type: 'small-list',
+				title: 'Sub Section 4',
+				items: [
+					{
+						key: 'sensei-share-your-grade',
+						extensionSlug: 'sensei-share-your-grade',
+					},
+					{
+						key: 'sensei-post-to-course',
+						extensionSlug: 'sensei-post-to-course',
+					},
+				],
+			},
+		],
+	},
+];
 
-	return (
-		<>
-			{ featuredExtensions.length > 0 && (
-				<section className="sensei-extensions__section sensei-extensions__grid__col --col-12">
-					<h2 className="sensei-extensions__section__title">
-						{ __( 'Featured', 'sensei-lms' ) }
-					</h2>
-					<ul className="sensei-extensions__section__content sensei-extensions__featured-list">
-						{ featuredExtensions.map( ( extension ) => (
+/**
+ * Renders the sections based on the skeleton structure. It can also render subsections recursively.
+ *
+ * @param {Array}  skeleton         Skeleton to be rendered.
+ * @param {Object} extensionsBySlug Extensions by slug to be rendered.
+ */
+const renderSections = ( skeleton, extensionsBySlug ) =>
```

Comment:
We could resolve the extensions in the `skeleton` structure as a separate step, then the renderer wouldn't need to pass this additional `extensionsBySlug` lookup argument around. 

For naming, instead of `skeleton`, would `layout` or even `sections` be more accurate?

Reply:
> We could resolve the extensions in the skeleton structure as a separate step, then the renderer wouldn't need to pass this additional extensionsBySlug lookup argument around.

If I understood correctly, you mean resolving the layout adding all the extension objects in the place of the slugs, right? If it's that, it would increase the complexity because we'd need to make the recursion twice.

> For naming, instead of skeleton, would layout or even sections be more accurate?

Renamed here: https://github.com/Automattic/sensei/pull/4197/commits/cfe89eb3dcc64b975500722be9fd68b09563672d